test/yagna/module/payments/test_zero_amount_txs.py

```diff
@@ -0,0 +1,161 @@
+        provider = runner.get_probes(probe_type=ProviderProbeWithLogSteps)[0]
+
+        # Market
+
+        await provider.wait_for_offer_subscribed()
+
+        task_package = task_package_template.format(
+            web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+        )
+
+        subscription_id, demand = await requestor.subscribe_template_demand(
+            task_package, demand_constraints
+        )
+
+        proposal = (
+            await requestor.wait_for_proposals(
+                subscription_id,
+                [provider],
+                lambda proposal: (
+                    proposal.properties.get("golem.runtime.name") == "wasmtime"
+                ),
+            )
+        )[0]
+        logger.info("Collected the proposal")
+
+        logger.info("Processing proposal from %s", provider.name)
+
+        # HINT: According to @kubam this is needed.
+        counterproposal_id = await requestor.counter_proposal(
+            subscription_id, demand, proposal
+        )
+        await provider.wait_for_proposal_accepted()
+
+        new_proposals = await requestor.wait_for_proposals(subscription_id, (provider,))
+        new_proposal = new_proposals[0]
+        assert new_proposal.prev_proposal_id == counterproposal_id
+
+        # Here: Agreement
+        agreement_id = await requestor.create_agreement(new_proposal)
+        await requestor.confirm_agreement(agreement_id)
+        await provider.wait_for_agreement_approved()
+
+        await requestor.unsubscribe_demand(subscription_id)
+        logger.info("Got agreement")
+
+        #  Zero-amount invoice is issued when agreement is terminated without activity
+        await wait_for_agreement_termination(requestor, provider, agreement_id)
+
+        # Payment
+
+        await provider.wait_for_invoice_sent()
+        invoices = await requestor.gather_invoices(agreement_id)
+        await requestor.pay_invoices(invoices)
+        await provider.wait_for_invoice_paid()
+
+        # verify requestor's invoice is settled
+        invoice = (await requestor.gather_invoices(agreement_id))[0]
+        assert invoice.amount == "0"
+        assert invoice.status == InvoiceStatus.SETTLED
+
+        # verify the provider invoice is settled
+        # TODO: Expose PaymentOperationMixin from Provider-based probe
+        # pseudo-code:
+        # invoice2 = (await providers.gather_invoices(agreement_id))[0]
+        # assert invoice2.amount == "0"
+        # assert invoice2.status == InvoiceStatus.SETTLED
+
+
+async def wait_for_agreement_termination(requestor, provider, agreement_id):
+    """Wait for agreement termination with retries in a given timespan."""
+
+    logger.info("Waiting for the agreement termination... ")
+    total_retry_time_seconds = 30
+    start = time.time()
+    while True:
+        assert total_retry_time_seconds > time.time() - start, "Retries time exceeded"
```

Comment:
How about that?
```suggestion
    deadline = time.time() + 30
    while True:
        assert time.time() < deadline, "Retries time exceeded"
```

Reply:
I think I will name it `timeout_seconds`, but I like an idea!

Reply:
I think "dealine" would be more appropriate because it's a _point in time_ while "timeout" is used to refer to a _period of time_ but let's not be fussy.

Reply:
UPDATE: Mentioned code is gone as I try `requestor.wait_for_approval(agreement_id)`